ale_linters/c/checkpatch.vim

```diff
@@ -0,0 +1,59 @@
+" Author: Matt Ihlenfield, mtihlenfield@protonmail.com
+" Description: checkpatch linter for c files
+
+let s:default_options = '--strict --no-tree'
+
+" Set the  location of the checkpatch executable. By default it assumes it
+" is in $(pwd)/scripts/checkpatch.pl
+call ale#Set('c_checkpatch_executable', '')
+
+call ale#Set('c_checkpatch_options', s:default_options)
+
+function! ale_linters#c#checkpatch#GetExecutable(buffer) abort
+    let l:checkpatch_executable = ale#Var(a:buffer, 'c_checkpatch_executable')
+
+    if l:checkpatch_executable  is# ''
+        let l:checkpatch_executable = fnamemodify('./scripts/checkpatch.pl', ':p')
```

Comment:
`.` can fail, because the working directory is configurable and can change when you switch buffers. Use `ale#path#FindNearestFile(a:buffer, 'scripts/checkpatch.pl')` to find the script instead. Disable searching for this executable when `g:ale_use_global_executables` is `0`. See `:help g:ale_use_global_executables`. Add tests for all of this. You can create empty dummy files in subdirectories. See other tests.